lib/model/image.py

```diff
@@ -30,8 +79,10 @@ def get_iobytes_for_image(self, image: schemas.Message) -> io.BytesIO:
             ).read()
         )
 
-    def process(self, image: schemas.Message) -> schemas.GenericItem:
+
+    def process(self, image: schemas.Message) -> schemas.ImageItem:
         """
         Generic function for returning the actual response.
         """
-        return self.compute_pdq(self.get_iobytes_for_image(image))
+        img_iobytes = self.get_iobytes_for_image(image)
+        return {"hash_value": self.compute_pdq(img_iobytes), "sscd_value": self.compute_sscd(img_iobytes)}
```

Comment:
Let's put try-except error catching around compute_pdq and compute_sscd . If one of the two fails, I'd like to still be able to return the other

Reply:
Ahmed and I discussed putting some logging into this function to track how much time is spent on SSCD and PDQ. We'll also pre-test on co-lab to have an approximate idea so we know how much additional work we're taking on by running SSCD for every image request

```

pdq=None
try:
	pdq_time=Time.now()
	pdq=self.compute_pdq(img_iobytes)
except:
	...
finally:
	pdq_time=Time.now()-pdq_time
	log....(f"PDQ took {pdq_time} seconds to process {filename}")
	

#Similar for sscd=self.compute_sscd(img_iobytes)

return {"hash_value": pdq, "sscd_value": sscd}
```

Reply:
I added a generic except as I can't tell what sort of errors might happen during pdq or sscd calculation. I also calculated the processing time in the `try` clause not `finally` as if the calculation failed, we don't need to log the processing time.